### PR TITLE
Fixes for LLVM 9

### PR DESCRIPTION
**Pull request by sklam**

### CODE REVIEW

File: numba/core/codegen.py [hunk 1]
```diff
@@ -829,6 +829,8 @@ def _customize_tm_options(self, options):
         arch = ll.Target.from_default_triple().name
         if arch.startswith('x86'): # one of x86 or x86_64
             reloc_model = 'static'
+        elif arch.startswith('ppc'):
+            reloc_model = 'pic'
```
Comment (stuartarchibald) [inline numba/core/codegen.py:833]: Perhaps add a note here about how this is entirely different to that in LLVM's `getEffectiveRelocModel` etc itself, but seems to work. I had another vague recollection about this yesterday regarding PIC leading to giant amounts of mapped memory on PPC for LLVM 7/8, worth taking a look?

File: numba/core/runtime/nrtopt.py [hunk 2]
```diff
@@ -9,7 +9,17 @@
...
-    r'([\'"]?[-a-zA-Z$._][-a-zA-Z$._0-9]*[\'"]?:)|^define|^;\s*<label>')
+    r'|'.join([
+        # unamed BB is just a plain number
+        r'[0-9]+:',
+        # with a proper identifer (see llvm langref)
+        r'[\'"]?[-a-zA-Z$._0-9][-a-zA-Z$._0-9]*[\'"]?:',
+        # is a start of a function definition
+        r'^define',
+        # no name
+        r'^;\s*<label>',
```
Comment (stuartarchibald) [inline numba/core/runtime/nrtopt.py:20]: thanks for adding these comments

**Comment by sklam** [comment]
Patch up CI to use `.label`

File: numba/core/codegen.py (hunk 1 above)
Comment (sklam) [reply numba/core/codegen.py:833]: I ran the full testsuite on PPC and it passes basically everything. Is that a good sign?

**Comment by sklam** [comment]
linux & mac has passed CI
windows llvmlite build is stuck.
not yet have aarch64 llvmdev
ppc64le have passed manual testing of numba

**Comment by sklam** [comment]
```Invalid PPC CTR loop!
[2020-05-29 15:25:33,845] {docker_operator.py:262} INFO - UNREACHABLE executed at /opt/conda/conda-bld/llvmdev_1583256883192/work/lib/Target/PowerPC/PPCCTRLoops.cpp:214!
[2020-05-29 15:25:33,845] {docker_operator.py:262} INFO - Fatal Python error: Aborted

File: numba/core/codegen.py (hunk 1 above)
Comment (stuartarchibald) [reply numba/core/codegen.py:833]: Yes!

**Comment by sklam** [comment]
Reverted changes that made this PR test against LLVM 9.

**Comment by sklam** [comment]
This PR needs to be merged as soon as https://github.com/numba/llvmlite/pull/548 is merged

**Comment by stuartarchibald** [comment]
Think this needs a change similar to https://github.com/numba/numba/pull/3437/files#diff-caa68793dddeb30ff0cf094862c99d17R61-R62 making?

**Pull request merged**